benchexec/tools/ultimate.py

```diff
@@ -204,113 +188,125 @@ def _requires_ultimate_data(self, executable):
-                            "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
-                                self.api, self.version(executable)
-                            )
+                    )
+            elif "-ultimatedata" in options and "-data" not in options:
+                if self.api == 2:
+                    cmdline += ["-data", "@noDefault"]
+                if self.api == 1:
+                    raise ValueError(
+                        "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
+                            self.api, self.version(executable)
                         )
-            else:
-                if "-data" not in options:
-                    if self.api == 2 or self.api == 1:
-                        cmdline += [
-                            "-data",
-                            os.path.join(os.path.dirname(executable), "data"),
-                        ]
-
-            cmdline += options
-
-            if tasks:
-                cmdline += ["-i"] + tasks
-            self.__assert_cmdline(
-                cmdline,
-                "cmdline contains empty or None argument when using Ultimate raw mode: ",
-            )
-            return cmdline
-
-        # there is no way to run ultimate; not enough parameters
-        raise UnsupportedFeatureException(
-            "Unsupported argument combination: options={} propertyfile={} rlimits={}".format(
-                options, propertyfile, rlimits
-            )
+                    )
+        else:
+            if "-data" not in options:
+                if self.api == 2 or self.api == 1:
+                    cmdline += [
+                        "-data",
+                        os.path.join(os.path.dirname(executable), "data"),
+                    ]
+
+        cmdline += options
+
+        if task.input_files_or_empty:
+            cmdline += ["-i", *task.input_files]
+        self.__assert_cmdline(cmdline, "No_Wrapper")
+        return cmdline
+
+    def _cmdline_default(self, executable, options, task):
+        # use the old wrapper script if a property file is given
+        cmdline = [executable, "--spec", task.property_file]
+        if task.input_files_or_empty:
+            cmdline += ["--file", *task.input_files]
+        cmdline += options
+        self.__assert_cmdline(cmdline, "Default")
+        return cmdline
+
+    def _cmdline_svcomp17(self, executable, options, task):
+        cmdline = [executable, task.property_file]
+        cmdline += [
+            option for option in options if option not in _SVCOMP17_FORBIDDEN_FLAGS
+        ]
+        cmdline.append("--full-output")
+        cmdline += task.input_files
+        self.__assert_cmdline(cmdline, "SVCOMP17")
+        return cmdline
+
+    @staticmethod
+    def __assert_cmdline(cmdline, mode):
+        assert all(
+            cmdline
+        ), "cmdline contains empty or None argument when using %s mode: %s".format(
```

Comment:
`.format()` does not support `%s`, so this needs to changed:
```suggestion
        ), "cmdline contains empty or None argument when using {} mode: {}".format(
```